src/java.base/share/classes/java/lang/System.java

```diff
@@ -114,15 +114,24 @@ private System() {
      * The "standard" input stream. This stream is already
      * open and ready to supply input data. Typically this stream
      * corresponds to keyboard input or another input source specified by
-     * the host environment or user.
+     * the host environment or user. In case this stream is wrapped
+     * in a {@link java.io.InputStreamReader}, {@link Console#charset()}
+     * should be used for the charset, or consider using
+     * {@link Console#reader()}.
+     *
+     * @see Console#charset()
+     * @see Console#reader()
```

Comment:
What would you think about changing the example in InputStreamReader class description as part of this?

Reply:
Replaced `System.in` with generic `anInputStream`, as changing `new InputStreamReader` with `Console.reader()` would defy the purpose of the example.